File: BEmuJava/bemu/src/main/java/com/bemu/BEmu/IntradayBarRequest/RequestIntradayBarElementString.java
//------------------------------------------------------------------------------
// <copyright project="BEmuJava" file="BEmu.IntradayBarRequest.RequestIntradayBarElementString.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bemu.BEmu.IntradayBarRequest;

import com.bemu.BEmu.Element;
import com.bemu.BEmu.Name;
import com.bemu.BEmu.Schema;

import java.lang.StringBuilder;

public class RequestIntradayBarElementString extends Element
{
    private final String _elementName, _value;

    public RequestIntradayBarElementString(String elementName, String value)
    {
        this._elementName = elementName;
        this._value = value;
    }
    
    public Name name() throws Exception
    {
    	return new Name(this._elementName);
    }
    
    public int numValues()
    {
    	return 1;
    }
    
    public int numElements()
    {
    	return 0;
    }

    public Schema.Datatype datatype()
    {
    	return Schema.Datatype.STRING;
    }
    
    protected String getSecurity()
    {
    	return this._value;
    }
    
    public Object getValue()
    {
        return this._value;
    }
    
    public String getValueAsString()
    {
        return this._value;
    }
    
    public String getValueAsString(int index) throws Exception
    {
        if (index == 0)
            return this._value;
        else
            return super.getValueAsString(index);
    }
    
    protected StringBuilder prettyPrint(int tabIndent)
    {
        String tabs = com.bemu.BEmu.types.IndentType.Indent(tabIndent);
        StringBuilder result = new StringBuilder();

        result.append(String.format("%s%s = %s%s", tabs, this._elementName, this._value, System.getProperty("line.separator")));

        return result;
    }

}

File: BEmu/IntradayBarRequest/RequestIntradayBarElementString.cs
//------------------------------------------------------------------------------
// <copyright project="BEmu" file="IntradayBarRequest/RequestIntradayBarElementString.cs" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

namespace BEmu.IntradayBarRequest
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;

    internal class RequestIntradayBarElementString : Element
    {
        private readonly string _elementName, _value;

        internal RequestIntradayBarElementString(string elementName, string value)
        {
            this._elementName = elementName;
            this._value = value;
        }

        public override int NumElements { get { return 0; } }
        public override int NumValues { get { return 1; } }
        public override Name Name { get { return new Name(this._elementName); } }
        public override bool IsArray { get { return false; } }
        public override bool IsComplexType { get { return false; } }

        internal string Security { get { return this._value; } }

        internal override StringBuilder PrettyPrint(int tabIndent)
        {
            return base.PrettyPrintHelper(tabIndent, this._value.ToString());
        }

        public override object GetValue()
        {
            return this._value;
        }

        public override string GetValueAsString()
        {
            return this._value;
        }

        public override string GetValueAsString(int index)
        {
            if (index == 0)
                return this._value;
            else
                return base.GetValueAsString(index);
        }

    }
}
